GitUI/Editor/FileViewer.cs

```diff
@@ -17,14 +17,18 @@
 using GitUI.Properties;
 using GitUIPluginInterfaces;
 using JetBrains.Annotations;
-using Microsoft.VisualStudio.Threading;
 using ResourceManager;
 
 namespace GitUI.Editor
 {
     [DefaultEvent("SelectedLineChanged")]
     public partial class FileViewer : GitModuleControl
     {
+        /// <summary>
+        /// Raised when the Escape key is pressed (and only when no selection exists, as the default behaviour of escape is to clear the selection).
+        /// </summary>
+        public event Action EscapePressed;
```

Comment:
[MSDN guidelines are to use `EventHandler`](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/event)

Reply:
Sorry for the delay. Haven't had much free time this week so far.

I've changed it, but allow me to go on the record and say that rule is dumb! Especially for non-library code. Using `EventHandler<EventArgs>` and `EventArgs.Empty` won't protect against breaking changes. And the fact that `sender` is always an object leads to dodgy casts and information sharing.

Reply:
I hear you. I mainly come from the consistency perspective.

Btw could you ask your peers perhaps what's their view on `event Action` vs `event EventHandler`?